editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/util/JobWaiterUtil.java

```diff
@@ -18,21 +18,34 @@
 
 public class JobWaiterUtil {
 	
+	boolean isEndless = false;
+	
 	private Throwable exception = null;
 
 	public void waitBuildAndRefreshJob(IProgressMonitor monitor) {
 		waitBuildAndRefreshJob(20, monitor);
 	}
 	
 	private void waitBuildAndRefreshJob(int decreasingCounter, IProgressMonitor monitor) {
-		if (decreasingCounter <= 0) {
-			if(exception != null){
-				ProjectTemplatesActivator.pluginLog().logWarning(exception);
-			} else {
-				ProjectTemplatesActivator.pluginLog().logWarning("Waiting for build to finish unsuccessfully.");
+		int currentCounter;
+		if(isEndless){
+			currentCounter = decreasingCounter;
+			ProjectTemplatesActivator.pluginLog().logInfo("log trace to ensure it is not looping");
+		} else {
+			if (decreasingCounter <= 0) {
+				if(exception != null){
+					ProjectTemplatesActivator.pluginLog().logError(exception);
+				} else {
+					ProjectTemplatesActivator.pluginLog().logError("Waiting for build to finish unsuccessfully.");
```

Comment:
in this else case we have no exception. so the warning or an info would be sufficient.